client/CVideoHandler.cpp

```diff
@@ -138,18 +133,24 @@ bool CVideoPlayer::open(std::string fname, bool loop, bool useOverlay, bool scal
 		// No video stream in that file
 		return false;
 
-	// Get a pointer to the codec context for the video stream
-	codecContext = format->streams[stream]->codec;
-
 	// Find the decoder for the video stream
-	codec = avcodec_find_decoder(codecContext->codec_id);
+	codec = avcodec_find_decoder(format->streams[stream]->codecpar->codec_id);
 
 	if (codec == nullptr)
 	{
 		// Unsupported codec
 		return false;
 	}
 
+	codecContext = avcodec_alloc_context3(codec);
+	// Get a pointer to the codec context for the video stream
+	int ret = avcodec_parameters_to_context(codecContext, format->streams[stream]->codecpar);
+	if (ret < 0)
+	{
+		//We cannot get codec from parameters
+		return false;

```

Comment:
Do we need to close codeContext here?

Reply:
I see it will be released by CVideoPlayer destructor anyway

Reply:
Yes, because we allocated context for new ffmpeg earlier. Of we do not close this context, we will get a memory leak.

Reply:
I will add it in next commit, thanks.

Reply:
Really looks like it is not needed. Next open will close all previously opened things.

Reply:
nullkiller, I added it in next commit, to avoid confusion. I can remove it again.

Reply:
Ha-ha, now we have another problem. As far as I understood we can use ffmpeg 5 for macos now? Could you change CI\mac\before_install.sh to use ffmpeg@5?

Reply:
Yes, you can, if there is ffmpeg 5 for Mac. It is you cancelled a pipelines?

Reply:
I will change.

Reply:
No, mac released ffmpeg@5 formulae and @4 is now keg-only as they say. And it is not enough now to just install it. So lets use ffmpeg5 for mac.

Reply:
I gave you the path of the file where to fix it

Reply:
Done, let's wait for checks.